experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -162,47 +164,89 @@ func checkInclusionProof(logEntryAnon *models.LogEntryAnon, registry strfmt.Regi
+	}
+
+	return nil
+}
+
 // checkEntryPubKeyMatchesExpectedKey compares the public key of the product
 // team in the Rekor log entry to the key of the product team passed as an
 // input to this wrapper. It returns an error if they are not equal
 // (or if valid keys could not be constructed)
-func checkEntryPubKeyMatchesExpectedKey(rekordEntry *rekord.V001Entry, prodTeamKeyBytes []byte) error {
-	logECDSAPubKey, err := pubKeyBytesToECDSA(rekordEntry.RekordObj.Signature.PublicKey.Content)
+func checkEntryPubKeyMatchesExpectedKey(rekorEntry *rekord.V001Entry, prodTeamKeyBytes []byte) error {
+	logECDSAPubKey, err := pubKeyBytesToECDSA(rekorEntry.RekordObj.Signature.PublicKey.Content)
 	if err != nil {
-		return fmt.Errorf("Invalid product team key in rekor log entry: %v", logECDSAPubKey)
+		return fmt.Errorf("invalid product team key in rekor log entry: %v", logECDSAPubKey)
 	}
 	prodTeamECDSAPubKey, err := pubKeyBytesToECDSA(prodTeamKeyBytes)
 	if err != nil {
-		return fmt.Errorf("Invalid product team public key passed as input: %v", logECDSAPubKey)
+		return fmt.Errorf("invalid product team public key passed as input: %v", logECDSAPubKey)
 	}
 	// Comparing the bytes of the public keys is not sufficient to check for key
 	// equality. Keys are considered equal if they are the same on the elliptic
 	// curve. Therefore, they could have different bytes, but still be the same
 	// key. The implementation of Equal from the ecdsa package checks this
 	// correctly, so it is used.
 	if !logECDSAPubKey.Equal(prodTeamECDSAPubKey) {
-		return fmt.Errorf("Input product team key does not match rekor log entry key: %v, %v", logECDSAPubKey, prodTeamECDSAPubKey)
+		return fmt.Errorf("input product team key does not match rekor log entry key: %v, %v", logECDSAPubKey, prodTeamECDSAPubKey)
 	}
 	return nil
 }
 
 // compareEndorsementAndRekordHash compares the hash referenced in a
-// rekord entry to the hash of input bytes.  For this use-case, the input
+// rekor entry to the hash of input bytes.  For this use-case, the input
 // bytes will be an endorsement file.
-func compareEndorsementAndRekorHash(rekordEntry *rekord.V001Entry, endorsementBytes []byte) error {
+func compareEndorsementAndRekorHash(rekorEntry *rekord.V001Entry, endorsementBytes []byte) error {
+	// The choice of hashing algorithm depends on the hashing algorithm used
+	// within the rekor entry which is defined in the structure
+	// RekordV001SchemaDataHash defined in rekord_v001_schema.go of the rekor
+	// source. This data structure uses a string to identify the hashing
+	// algorithm, but at the time of writing a comment in this code says:
+	// "Enum: [sha256]" above the algorithm part of the structure,
+	// suggesting that sha256 is the only supported choice.
 	endorsementHash := fmt.Sprintf("%x", sha256.Sum256(endorsementBytes))
-	if endorsementHash != *rekordEntry.RekordObj.Data.Hash.Value {
+	if endorsementHash != *rekorEntry.RekordObj.Data.Hash.Value {
```

Comment:
I think here, it makes sense to check that `rekorEntry.RekordObj.Data.Hash.Algorithm` is `sha256`, and return an `unsupported hash algorithm` error, if it is not.

Reply:
fixed